### PR TITLE
sync/atomic audit

### PR DESCRIPTION
**Pull request by jmank88**
https://app.clubhouse.io/chainlinklabs/story/15896/remove-use-of-sync-atomic-and-replace-with-mutexes-unless-we-are-absolutely-sure-it-is-safe

### CODE REVIEW

File: core/services/fluxmonitorv2/integrations_test.go [hunk 1]
```diff
@@ -432,9 +432,9 @@ func TestFluxMonitor_Deviation(t *testing.T) {
 	type k struct{ latestAnswer, updatedAt string }
 	expectedMeta := map[k]int{}
 
-	reportPrice := int64(100)
+	reportPrice := atomic.NewInt64(100)
```
Comment (jmank88) [inline core/services/fluxmonitorv2/integrations_test.go:435]: These `reportPrice` vars had a mix of atomic and non-atomic access.

**Comment by jmank88** [comment]
None of the other uses of `package atomic` look problematic to me, but swapping in the uber atomic package could still be valuable for readability and to prevent future mistakes. The std lib may be going in that direction to: https://research.swtch.com/gomm#maybe

File: core/services/fluxmonitorv2/integrations_test.go (hunk 1 above)
Comment (PiotrTrzpil) [reply core/services/fluxmonitorv2/integrations_test.go:435]: ah nice find! yeah that's likely part of what they warn about in the docs on using sync/atomic...

**Review by PiotrTrzpil — APPROVED** [review]

File: core/services/log/broadcaster_test.go [hunk 2]
```diff
@@ -167,24 +165,24 @@ func TestBroadcaster_BackfillOnNodeStartAndOnReplay(t *testing.T) {
...
-		times := atomic.LoadInt64(backfillCountPtr)
+		times := backfillCount.Load()
 		if times == 0 {
 			require.Equal(t, lastStoredBlockHeight-maxNumConfirmations-int64(blockBackfillDepth), fromBlock)
 		} else if times == 1 {
 			require.Equal(t, replayFrom, fromBlock)
 		}
 
-		atomic.StoreInt64(backfillCountPtr, times+1)
+		backfillCount.Store(times + 1)
```
Comment (jmank88) [inline core/services/log/broadcaster_test.go:175]: Do these `checkFilterLogs` funcs ever execute concurrently? If so, the ones incrementing via a `Store` call at the end will race.

Comment (jmank88) [reply core/services/log/broadcaster_test.go:175]: The first CI failure was the related test `TestBroadcaster_BackfillALargeNumberOfLogs` but it passed on a re-run, which suggests this is a race - fix incoming.

The second CI failure was an unrelated test `TestIntegration_DirectRequest`.

Comment (PiotrTrzpil) [reply core/services/log/broadcaster_test.go]: so the TestBroadcaster_BackfillALargeNumberOfLogs i just fixed in https://github.com/smartcontractkit/chainlink/pull/4950 
and DirectRequest is actually an indication of a non-test issue

Comment (jmank88) [reply core/services/log/broadcaster_test.go]: `TestIntegration_DirectRequest` is being addressed here: https://github.com/smartcontractkit/chainlink/pull/4956 and CI ran `TestBroadcaster_BackfillALargeNumberOfLogs` successfully twice now.

**Comment by jmank88** [comment]
I converted a few more similar cases to uber. Now there are 10 remaining `sync/atomic` vars, and they are documented as `// atomic`. Any opinions on converting them all to uber, so there is just one simple/safe standard?

**Comment by archseer** [comment]
I'd prefer to convert them all to avoid any future misuse 👍🏻

Is there a linter we could use to deny `sync/atomic` imports with a note to use uber atomic? Was looking at https://golangci-lint.run/usage/linters/#forbidigo potentially

**Review by archseer — DISMISSED** [review]

**Review by archseer — APPROVED** [review]

**Pull request merged**